src/main/java/com/sourcegraph/webhook/WebhookListener.java

```diff
@@ -0,0 +1,74 @@
+package com.sourcegraph.webhook;
+
+import com.atlassian.bitbucket.event.ApplicationEvent;
+import com.atlassian.bitbucket.event.backup.BackupStartedEvent;
+import com.atlassian.bitbucket.event.permission.PermissionEvent;
+import com.atlassian.bitbucket.event.pull.*;
+import com.atlassian.bitbucket.repository.Repository;
+import com.atlassian.bitbucket.scm.event.ScmEvent;
+import com.atlassian.event.api.AsynchronousPreferred;
+import com.atlassian.event.api.EventListener;
+import com.google.common.base.Joiner;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.sourcegraph.webhook.registry.Webhook;
+import com.sourcegraph.webhook.registry.WebhookRegistry;
+
+import javax.inject.Named;
+import java.util.*;
+
+@AsynchronousPreferred
+@Named("WebhookListener")
+public class WebhookListener {
+    private static Map<Class<?>, List<String>> triggers = new HashMap<>();
+
+    private static void register(Class<?> type, String key) {
+        String[] split = key.split(":");
+        List<String> pieces = new ArrayList<>();
+        for (int i = 0, index = 0; i < split.length; i++) {
+            index += split[i].length();
+            pieces.add(key.substring(0, index));
+            index++;
```

Comment:
This bit of code is quite non-obvious to me. It seems that we already have pieces in the `split` string array. Is this loop simply converting that type to a `List<String>`? Is there a simpler way to do that?

Reply:
The loop converts the input string into a List<String> containing all the prefixes. So `pr:comment:added` becomes `[pr, pr:comment, pr:comment:added]`. This is so if a webhook is registered for `pr:comment`, it will handle all the sub events: `pr:comment:added`, `pr:comment:edited`, `pr:comment:deleted`. `pr` would handle all pull request events.

Reply:
Cool. Additionally, instead of `pieces`, you could name the variable `prefixes` for added clarity.